Packet++/src/IPv4Layer.cpp

```diff
@@ -237,7 +237,7 @@ IPv4Layer& IPv4Layer::operator=(const IPv4Layer& other)
 void IPv4Layer::parseNextLayer()
 {
 	size_t hdrLen = getHeaderLen();
-	if (m_DataLen <= hdrLen)
+	if (m_DataLen <= hdrLen || hdrLen <= 0)

```

Comment:
`hdrLen` can't be lower than 0 because it's of type `size_t` which is unsigned.

So you can change this to:
```cpp
if (m_DataLen <= hdrLen || hdrLen == 0)
```